tracing-distributed/src/telemetry_layer.rs

```diff
@@ -237,7 +237,7 @@ where
                         parent_id: Some(self.trace_ctx_registry.promote_span_id(parent_id)),
                         initialized_at,
                         meta: event.metadata(),
-                        service_name: &self.service_name,
```

Comment:
This de-reference looks backwards incompatible. Is this change necessary to improve error handling?

Reply:
@jfhbrook-at-work This change is not necessary for the error handling. It's a warning that was only fixed since CI complained about it. I had not thought about backwards-compatibility, so thanks for  pointing this out.

As a warning: this same change is also in #12 for the same reason.

Reply:
Thanks for the heads up! I'll keep an eye on it.